samza-core/src/main/java/org/apache/samza/coordinator/JobCoordinator.java

```diff
@@ -57,10 +57,11 @@
   boolean awaitStart(long timeoutMs) throws InterruptedException;
   /**
    * Returns the logical ID assigned to the processor
-   * It is up to the user to ensure that different instances of StreamProcessor within a job have unique processor ID.
-   * @return integer representing the logical processor ID
+   * It is the responsibility of the JobCoordinator to assign a unique identifier based on the underlying environment.
+   *
+   * @return String representing a unique logical processor ID
    */
-  int getProcessorId();
+  String getLocalProcessorId();
```

Comment:
> JobCoordinator's APIs are only called within local StreamProcessor, and there is no use case to call JobCoordinator to get processIds other than the StreamProcessor wrapping the JobCoordinator

I actually never even thought about the method for getting all processorIds through the JobCoordinator interface. That will be internal to the JC implementation and will not be used by any other interacting component, such as the StreamProcessor. Hence, I didn't find the documentation be confusing. 
I will try to clarify in the documentation that this interface container methods that are required for interacting with StreamProcessor.